File: appoointment_parent/appointment_queue/src/main/webapp/js/controller/projectController.js
//控制层
app.controller('projectController' ,function($scope,$controller,projectService) {

    $controller('baseController', {$scope: $scope});//继承

    //读取列表数据绑定到表单中
    $scope.findAll=function(){
        projectService.findAll().success(
            function(response){
                $scope.list=response;
            }
        );
    }
    
    // 分页查询品牌列表
	$scope.findPage = function(page,size){
		projectService.findPage().success(
			function(response){
				$scope.list = response.rows;// 当前页的数据
				$scope.paginationConf.totalItems = response.total;// 更新总记录数
			}
	)};

    // 修改状态
    $scope.updateStatus=function(proId, status){
        projectService.updateStatus(proId, status).success(
            function(response){
                if(response.success){
                    $scope.reloadList();//刷新列表
                }else{
                    alert(response.message);
                }
            }
        );
    }

    //保存
    $scope.save=function(){
        var serviceObject;//服务层对象
        if($scope.entity.proId!=null){//如果有ID
            serviceObject=projectService.updateStatus( $scope.entity ); //修改
        }else{
            serviceObject=projectService.add( $scope.entity  );//增加
        }
        serviceObject.success(
            function(response){
                if(response.success){
                    //重新查询
                    $scope.reloadList();//重新加载
                }else{
                    alert(response.message);
                }
            }
        );
    }

    //查询实体
    $scope.findOne=function(proId){
        projectService.findOne(proId).success(
            function(response){
                $scope.entity= response;
            }
        );
    }

    $scope.status=["","正常","暂停","停止"];
	
});

File: appoointment_parent/appointment_queue/src/main/java/com/attx/controller/ProjectController.java
package com.attx.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.attx.entity.PageResult;
import com.attx.entity.Result;
import com.attx.pojo.Project;
import com.attx.service.ProjectService;

@RestController
@RequestMapping("/project")
public class ProjectController {

	@Autowired
	private ProjectService projectService;

	@RequestMapping("/getProjectList")
	public List<Project> getProjectList() {
		return projectService.getProjectList();
	}

	/**
	 * 分页查询
	 * @param page
	 * @param size
	 * @return
	 */
	@RequestMapping("/findpage")
	public PageResult findPage(int page, int size) {

		return projectService.findPage(page, size);
	}
	
	/**
	 * 增加
	 * @param project
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(Project project) {
		try {
			projectService.add(project);
			return new Result(true, "增加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "增加失败");
		}
	}

	/**
	 * 修改状态:1正常2暂停3停止
	 * @param proId,status
	 * @return
	 */
	@RequestMapping("/updateStatus")
	public Result updateStatus(Integer proId, String status) {
		try {
			projectService.updateStatus(proId,status);
			return new Result(true, "修改状态成功！");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改状态失败！");
		}
	}

	/**
	 * 设置批数上限
	 */
	@RequestMapping("/updateAllGroup")
	public Result updateAllGroup(Integer proId,String allGroup){
		try {
			projectService.updateAllGroup(proId,allGroup);
			return new Result(true, "修改批数上限成功！");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改批数上限失败！");
		}
	}

	/**
	 * 获取实体
	 * 
	 * @param proId
	 * @return
	 */
	@RequestMapping("/findOne")
	public Project findOne(Integer proId) {
		return projectService.findOne(proId);
	}

	/**
	 * 批量删除
	 * @param ids
	 * @return
	 */
	@RequestMapping("/delete")
	public Result delete(Integer[] ids) {
		try {
			projectService.delete(ids);
			return new Result(true, "删除成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "删除失败");
		}
	}
}
